tests/mock_network/mod.rs

```diff
@@ -433,96 +426,118 @@ fn carry_out_parsec_pruning() {
+    poll_until(&env, &mut nodes, |nodes| {
+        nodes.iter().all(|node| node.inner.is_elder())
+    });
+
     let initial_parsec_versions = parsec_versions(&nodes);
 
-    let mut rng = env.new_rng();
     // Keeps polling and dispatching user data till trigger a pruning.
     let max_gossips = 1_000;
+    let mut all_parsec_versions_increased = false;
     for _ in 0..max_gossips {
         let event = gen_vec(&mut rng, 10_000);
         nodes.iter_mut().for_each(|node| {
             node.inner.vote_for_user_event(event.clone()).unwrap();
         });
-        poll_and_resend(&mut nodes);
+
+        let mut consensus_counter = 0;
+        poll_until(&env, &mut nodes, |nodes| {
+            consensus_reached(nodes, &event, nodes.len(), &mut consensus_counter)
+        });
 
         let new_parsec_versions = parsec_versions(&nodes);
         if initial_parsec_versions
             .iter()
-            .zip(new_parsec_versions.iter())
-            .all(|(vi, vn)| vi < vn)
+            .zip(new_parsec_versions)
+            .all(|(&vi, vn)| vi < vn)
         {
+            all_parsec_versions_increased = true;
             break;
         }
     }
 
-    let expected = initial_parsec_versions.iter().map(|v| v + 1).collect_vec();
-    let actual = parsec_versions(&nodes);
-    assert_eq!(expected, actual);
+    assert!(all_parsec_versions_increased);
 
     let node = create_node_with_contact(&env, &mut nodes[0]);
     nodes.push(node);
-
-    poll_and_resend(&mut nodes);
-
-    verify_invariant_for_all_nodes(&env, &mut nodes);
+    poll_until(&env, &mut nodes, |nodes| {
+        node_joined(nodes, nodes.len() - 1)
+    });
+    verify_invariants_for_nodes(&env, &nodes);
 }
 
-// The paused node does not participate until resumed, so we need enough elders to reach
-// consensus even without it.
-const NODE_PAUSE_AND_RESUME_PARAMS: NetworkParams = NetworkParams {
-    elder_size: 4,
-    safe_section_size: 4,
-};
-
 #[test]
 fn node_pause_and_resume_simple() {
-    let env = Environment::new(NODE_PAUSE_AND_RESUME_PARAMS);
-    let nodes = create_connected_nodes(&env, 2 * env.safe_section_size() - 2);
-    let new_node_id = FullId::gen(&mut env.new_rng());
-    node_pause_and_resume(env, nodes, new_node_id)
+    let env = Environment::new(NetworkParams {
+        elder_size: LOWERED_ELDER_SIZE,
+        safe_section_size: LOWERED_ELDER_SIZE + 1,
+    });
+
+    let mut nodes = create_connected_nodes(&env, env.safe_section_size());
+    let paused_state = pause_node_and_poll(&env, &mut nodes);
+
+    add_node_to_section(&env, &mut nodes, &Prefix::default());
+    poll_until(&env, &mut nodes, |nodes| {
+        node_joined(nodes, nodes.len() - 1)
+    });
+    let new_id = *nodes.last().unwrap().id();
+
+    nodes.push(TestNode::resume(paused_state));
+
+    // If the paused node is elder, verify it caught up to the new node joining.
+    if nodes.last().unwrap().inner.is_elder() {
```

Comment:
is it guaranteed this situation will happen during the test?
I mean, maybe we shall ensure the paused node is the last to ensure the check of caught up to the new node will be properly always?

Reply:
Do you mean is it guaranteed that the paused node is elder? If so, then no, it is not. That's why there is this `if`. So this tests needs to be run multiple times to make sure this situation is hit.

On the other hand, maybe we can pick the node to pause only from elders? But then we probably do want to test pausing also non-elders. Maybe we should have two tests?

Reply:
> Do you mean is it guaranteed that the paused node is elder? If so, then no, it is not.

Yes, I mean this.

> So this tests needs to be run multiple times to make sure this situation is hit.

Yes, this is where I concerned. For how many iterations will this situation be hit? I am concerned it may not be tested enough. Ideally, such situation shall be covered within every iteration.

Reply:
OK, I agree. But given that the whole pause/resume functionality is still not finished, this issue is probably not that critical right now. What I'd do is to create an issue to address this later. How does that sound?

Reply:
yeah, that will be ok.  :)

Reply:
Created the issue: https://github.com/maidsafe/routing/issues/2081